---
title: "Case-Study-for-ADO-PCE-1"
author: "Pío Sierra"
date: "30/4/2020"
output:
  html_document:
    df_print: paged
---

```{r setup, include=FALSE}
if(!(require(knitr))) 
require(knitr)
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, 
                      comment = NA, prompt = TRUE, tidy = FALSE, 
                      fig.width = 7, fig.height = 7, fig_caption = TRUE,
                      cache=FALSE)
if(!(require(BiocManager))) 
  install.packages("BiocManager")
if(!(require(Biobase))) 
  BiocManager::install("Biobase")
if(!(require(kableExtra))) 
  BiocManager::install("kableExtra")
if(!(require(oligo))) 
  BiocManager::install("oligo")
if(!(require(pvca))) 
  BiocManager::install("pvca")
if(!(require(ggrepel))) 
  install.packages("ggrepel")
if(!(require(colorspace))) 
  install.packages("colorspace")
if(!(require(gplots))) 
  install.packages("gplots")
if(!(require(RColorBrewer))) 
  install.packages("RColorBrewer")
if(!(require(ggplot2))) 
  install.packages("ggplot2")
if(!(require(hgu133plus2.db))) 
BiocManager::install("hgu133plus2.db")
if(!(require(arrayQualityMetrics))) 
  BiocManager::install("arrayQualityMetrics")
if(!(require(limma))) 
  BiocManager::install("limma")
if(!(require(genefilter))) 
  BiocManager::install("genefilter")
if(!(require(annotate))) 
  BiocManager::install("annotate")
if(!(require(ReactomePA))) 
  BiocManager::install("ReactomePA")
```

# Abstract

# Introducción

El estudio va a ser sobre
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE17400
Study:
https://www.ncbi.nlm.nih.gov/pubmed/20090954

Y procedemos a descargar los archivos.

Utilizamos los datos de 
Affymetrix Human Genome U133 Plus 2.0 Array annotation data (chip hgu133plus2)
http://bioconductor.org/packages/release/data/annotation/html/hgu133plus2.db.html


```{r eval = FALSE}

url <- "https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE17400&format=file"
utils::download.file(url, destfile = "data.tar", mode="wb") 
utils::untar("data.tar", exdir = "data")

```

## 1. Identificar que grupos hay y a qué grupo pertenece cada muestra.

"For statistical purposes, all 27 arrays were analyzed as 18 separate groups (mock-, SARS-CoV-infected, and DHOV-infected cells at 12, 24, and 48 hrs)." 

Following thar criteria we create 18 group, with the 9 different combinations an using the sets 1 and 2 for the new set "a" and the set 3 for the new set "b".

```{r}
targets <- read.csv2("targets.csv", header = TRUE, sep = ";") 
knitr::kable(targets, booktabs = TRUE,
             caption = 'Contenido de los target files utilizados para el análisis')
library(oligo)
celFiles <- list.celfiles("./data", full.names = TRUE, listGzipped=TRUE)
library(Biobase)
my.targets <-read.AnnotatedDataFrame(file.path(".","targets.csv"), 
                                     header = TRUE, row.names = 1, 
                                     sep=";") 
rawData <- read.celfiles(celFiles, phenoData = my.targets)
```
2. Control de calidad de los datos crudos

Análisis de calidad.

```{r QCRaw, message=FALSE, warning=FALSE, eval=FALSE}
require(arrayQualityMetrics)
arrayQualityMetrics(rawData, outdir = file.path("./results", "QCDir.Raw"), force=TRUE)
```
Análisis de componentes principales.

```{r}
plotPCA3 <- function (datos, labels, factor, title, scale,colores, size = 1.5, glineas = 0.25) {
  data <- prcomp(t(datos),scale=scale)
  # plot adjustments
  dataDf <- data.frame(data$x)
  Group <- factor
  loads <- round(data$sdev^2/sum(data$sdev^2)*100,1)
  # main plot
  p1 <- ggplot(dataDf,aes(x=PC1, y=PC2)) +
    theme_classic() +
    geom_hline(yintercept = 0, color = "gray70") +
    geom_vline(xintercept = 0, color = "gray70") +
    geom_point(aes(color = Group), alpha = 0.55, size = 3) +
    coord_cartesian(xlim = c(min(data$x[,1])-5,max(data$x[,1])+5)) +
    scale_fill_discrete(name = "Group")
  # avoiding labels superposition
  p1 + geom_text_repel(aes(y = PC2 + 0.25, label = labels),segment.size = 0.25, size = size) + 
    labs(x = c(paste("PC1",loads[1],"%")),y=c(paste("PC2",loads[2],"%"))) +  
    ggtitle(paste("Principal Component Analysis for: ",title,sep=" "))+ 
    theme(plot.title = element_text(hjust = 0.5)) +
    scale_color_manual(values=colores)
  }
colorsq <- brewer.pal(9, "Set1")
colorss <- brewer.pal(9, "YlOrRd")
colorsd <- brewer.pal(9, "Spectral")

plotPCA3(exprs(rawData), labels = targets$ShortName, factor = targets$Group, 
         title="Raw data", scale = FALSE, size = 3, 
         colores = c(rep(colorsq[1], 6),
                     rep(colorsq[2], 6),
                     rep(colorsq[3], 6)))
```
Observamos la distribución de intensidad de los arrays observando los boxplots.

```{r}

boxplot(rawData, cex.axis=0.5, las=2,  which="all", 
         col = c(rep(colorsd[1], 3), 
                 rep(colorsd[2], 3), 
                 rep(colorsd[3], 3),
                 rep(colorsd[4], 3),
                 rep(colorsd[5], 3),
                 rep(colorsd[6], 3),
                 rep(colorsd[7], 3),
                 rep(colorsd[8], 3),
                 rep(colorsq[9], 3)),
         main="Distribution of raw intensity values")



```

3. Normalización

```{r}
eset_rma <- rma(rawData)
```

4. [Control de calidad de los datos normalizados] (opcional)


```{r eval=FALSE}
arrayQualityMetrics(eset_rma, outdir = file.path("./results", "QCDir.Norm"), force=TRUE)
```
```{r}
plotPCA3(exprs(eset_rma), labels = targets$ShortName, factor = targets$Group, 
         title="Normalized data", scale = FALSE, size = 3, 
         colores = c(rep(colorsq[1], 6),
                     rep(colorsq[2], 6),
                     rep(colorsq[3], 6)))

boxplot(eset_rma, cex.axis=0.5, las=2,  which="all", 
         col = c(rep(colorsd[1], 3), 
                 rep(colorsd[2], 3), 
                 rep(colorsd[3], 3),
                 rep(colorsd[4], 3),
                 rep(colorsd[5], 3),
                 rep(colorsd[6], 3),
                 rep(colorsd[7], 3),
                 rep(colorsd[8], 3),
                 rep(colorsq[9], 3)),
         main="Distribution of raw intensity values")

```

5. Filtraje no específico [opcional]
6. Identificación de genes diferencialmente expresados

```{r}
sds <- apply (exprs(eset_rma), 1, sd)
sdsO<- sort(sds)
plot(1:length(sdsO), sdsO, main="Distribution of variability for all genes",
     sub="Vertical lines represent 90% and 95% percentiles",
     xlab="Gene index (from least to most variable)", ylab="Standard deviation")
abline(v=length(sds)*c(0.9,0.95))

annotation(eset_rma) <- "hgu133plus2.db"
filtered <- nsFilter(eset_rma, 
                     require.entrez = TRUE, remove.dupEntrez = TRUE,
                     var.filter=TRUE, var.func=IQR, var.cutoff=0.75, 
                     filterByQuantile=TRUE, feature.exclude = "^AFFX")
print(filtered$filter.log)
eset_filtered <-filtered$eset
write.csv(exprs(eset_rma), file="./results/normalized.Data.csv")
write.csv(exprs(eset_filtered), file="./results/normalized.Filtered.Data.csv")
save(eset_rma, eset_filtered, file="./results/normalized.Data.Rda")
```
```{r}
if (!exists("eset_filtered")) load (file="./results/normalized.Data.Rda")


require(limma)
designMat<- model.matrix(~0+Group, pData(eset_filtered))
colnames(designMat) <- sub("Group","", colnames(designMat))
print(designMat)

cont.matrix <- makeContrasts (D12vsM12 = ((DOHV12a+DOHV12b)-(Mock12a+Mock12b))-((Mock12a-Mock12b))*1.5,
                              D24vsM24 = ((DOHV24a+DOHV24b)-(Mock24a+Mock24b))-((Mock24a-Mock24b))*1.5,
                              D48vsM48 = ((DOHV48a+DOHV48b)-(Mock48a+Mock48b))-((Mock48a-Mock48b))*1.5,
                              S12vsM12 = ((SarsCoV12a+SarsCoV12b)-(Mock12a+Mock12b))-((Mock12a-Mock12b))*1.5,
                              S24vsM24 = ((SarsCoV24a+SarsCoV24b)-(Mock24a+Mock24b))-((Mock24a-Mock24b))*1.5,
                              S48vsM48 = ((SarsCoV48a+SarsCoV48b)-(Mock48a+Mock48b))-((Mock48a-Mock48b))*1.5,
                              M12=Mock12a-Mock12b,
                              M24=Mock24a-Mock24b,
                              M48=Mock48a-Mock48b,
                              D12vsM12t=((DOHV12a+DOHV12b)-(Mock12a+Mock12b)),
                              D24vsM24t=((DOHV24a+DOHV24b)-(Mock24a+Mock24b)),
                              D48vsM48t=((DOHV48a+DOHV48b)-(Mock48a+Mock48b)),
                              levels=designMat)
print(cont.matrix)
```
```{r}
cont.names <- c("D12vsM12","D24vsM24", "D48vsM48", "S12vsM12", "S24vsM24", "S48vsM48","M12", "M24", "M48", "D12vsM12t","D24vsM24t", "D48vsM48t")
cont.n <- length(cont.names)
require(limma)
fit<-lmFit(eset_filtered, designMat)
fit.main<-contrasts.fit(fit, cont.matrix)
fit.main<-eBayes(fit.main)
class(fit.main)
cont.list <- list()
for (i in 1:cont.n) {
  cont.list[[i]] <- topTable (fit.main, number=nrow(fit.main), coef=cont.names[i], adjust="fdr")
  print(cont.names[i])  
  print(head(cont.list[[i]]))
  print(tail(cont.list[[i]]))
}



```

```{r GeneAnnotation, message=FALSE, warning=FALSE}
annotatedTopTable <- function(topTab, anotPackage)
{
  topTab <- cbind(PROBEID=rownames(topTab), topTab)
  myProbes <- rownames(topTab)
  thePackage <- eval(parse(text = anotPackage))
  geneAnots <- select(thePackage, myProbes, c("SYMBOL", "ENTREZID", "GENENAME"))
  annotatedTopTab<- merge(x=geneAnots, y=topTab, by.x="PROBEID", by.y="PROBEID")
return(annotatedTopTab)
}
```

```{r annotateTopTables}
anno.list <- list()
for (i in 1:cont.n) {
  anno.list[[i]] <- annotatedTopTable(cont.list[[i]], anotPackage="hgu133plus2.db")

write.csv(anno.list[i], file=paste("./results/topAnnotated_",cont.names[i],".csv",sep=""))
}
```


```{r annotatedTop}
for (i in 1:cont.n) {
  print(paste('Annotations added to results topTable for the comparison: ',cont.names[[i]]))
  b <- anno.list[[i]]
  print(head(b[order(b$adj.P.Val),][1:5,1:4]))
}
```
```{r}
for (i in 1:cont.n) {
geneSymbols <- select(hgu133plus2.db, rownames(fit.main), c("SYMBOL"))
SYMBOLS<- geneSymbols$SYMBOL
volcanoplot(fit.main, coef=i, highlight=4, names=SYMBOLS, 
           main=paste("Differentially expressed genes", colnames(cont.matrix)[i], sep="\n"))
abline(v=c(-1,1))
}
```


7. Anotación de los resultados
8. Comparación entre distintas comparaciones (si hay más de una comparación, ver que genes han sido seleccionados en más de una comparación)
```{r}

res<-decideTests(fit.main, method="separate", adjust.method="fdr", p.value=0.05, lfc=1.5)
sum.res.rows<-apply(res,1,sum)
res.up<-res[sum.res.rows>0,] 
res.down<-res[sum.res.rows<0,]
res.selected<- res[sum.res.rows!=0,]
print(summary(res))

vennDiagram (res.up[,1:3], cex=0.9)
title("Genes in common between the three comparisons\n Genes up regulated with FDR < 0.05 and logFC > 1.5")
vennDiagram (res.down[,1:3], cex=0.9)
title("Genes in common between the three comparisons\n Genes down regulated with FDR < 0.05 and logFC > 1.5")
```

Heatmaps

```{r}
probesInHeatmap <- rownames(res.selected)
HMdata <- exprs(eset_filtered)[rownames(exprs(eset_filtered)) %in% probesInHeatmap,]

geneSymbols <- select(hgu133plus2.db, rownames(HMdata), c("SYMBOL"))
SYMBOLS<- geneSymbols$SYMBOL
rownames(HMdata) <- SYMBOLS
write.csv(HMdata, file = file.path("./results/data4Heatmap.csv"))
```




9. Análisis de significación biológica (“Gene Enrichment Analysis”)